src/Extension/Extension.php

```diff
@@ -203,16 +211,30 @@ public function setVersion($version)
      * @param array $extensionSet: An associative array where keys are the composer package names
      *                             of installed extensions. Used to figure out which dependencies
      *                             are flarum extensions.
+     * @param array $enabledIds:   An associative array where keys are the composer package names
+     *                             of enabled extensions. Used to figure out optional dependencies.
      */
-    public function calculateDependencies($extensionSet)
+    public function calculateDependencies($extensionSet, $enabledIds)
     {
         $this->extensionDependencyIds = (new Collection(Arr::get($this->composerJson, 'require', [])))
             ->keys()
+            ->merge(Arr::get($this->composerJson, 'extra.flarum-extension.optional-dependencies', []))
```

Comment:
What's the purpose of including the optional dependencies with the "hard" dependencies? If we already get the optional ones later, then merge the two in the order function, wouldn't it make sense to keep them separate?

Reply:
Ah, yeah that was a mistake